src/BotwinExtensions.cs

```diff
@@ -44,19 +40,24 @@ public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, Bo
                 foreach (var module in scope.ServiceProvider.GetServices<BotwinModule>())
                 {
                     var moduleType = module.GetType();
-                    var distinctPaths = module.Routes.Keys.Select(route => route.path).Distinct();
+
+                    var distinctPaths = module.Routes.Keys.Select(route =>
+                    {
+                        logger?.LogTrace($"Found and adding {moduleType.Name} with {route.verb} /{route.path} to routing");
+                        return route.path;
+                    }).Distinct();
 
                     foreach (var path in distinctPaths)
                     {
-                        routeBuilder.MapRoute(path, CreateRouteHandler(path, moduleType));
+                        routeBuilder.MapRoute(path, CreateRouteHandler(path, moduleType, logger));
                     }
                 }
             }
 
             return builder.UseRouter(routeBuilder.Build());
         }
 
-        private static RequestDelegate CreateRouteHandler(string path, Type moduleType)
+        private static RequestDelegate CreateRouteHandler(string path, Type moduleType, ILogger logger)
```

Comment:
I honestly don’t think you should use the same logger here. `CreateRouteHandler` basically creates a middleware. That middleware is run as part of the request pipeline. So it runs much later than any of the other things that logger (the `BotwinExtensions` logger!) is logging.

You should probably request a new logger inside of the middleware. And you should probably request an `ILogger<moduleType>` there so that people can actually figure out what module runs.